lib/pushy_client/protocol_handler.rb

```diff
@@ -411,11 +404,10 @@ def check_server_address(name, server_address)
     # MAX_BODY_SIZE. If they do, log a warning and return the params without the
     # stdout and stderr keys. If they do not, return the params hash unmodified.
     def validate_params(params = {})
-      stdout_bytes = params[:stdout].to_s.bytesize
-      stderr_bytes = params[:stderr].to_s.bytesize
-
-      if (stdout_bytes + stderr_bytes) > MAX_BODY_SIZE
-        Chef::Log.warn("Command output too long. Will not be sent to server.")
+      if (params.to_s.bytesize) > client.max_body_size.to_i
```

Comment:
So params.to_s.bytesize measures the size of ruby hash string representation, not the size of the json packet; that will use a different encoding and will not match. (for example, the symbol keys end up quoted in the json representation)

Reply:
I have created separate methods for the validate and regenerate json.